Client/mods/deathmatch/logic/CClientModel.cpp

```diff
@@ -78,30 +78,49 @@ bool CClientModel::Allocate(ushort usParentID)
     return false;
 }
 
+// You can call it only in destructor for DFF.
 bool CClientModel::Deallocate()
 {
     if (!m_bAllocatedByUs)
         return false;
+
+    SetParentResource(nullptr);
+
     CModelInfo* pModelInfo = g_pGame->GetModelInfo(m_iModelID, true);
     if (!pModelInfo || !pModelInfo->IsValid())
         return false;
 
-    SetParentResource(nullptr);
+    if (m_eModelType != eClientModelType::TXD)
+    {
+        // Remove model info
+        pModelInfo->DeallocateModel();
+    }
+
+    return true;
+}
+
+void CClientModel::RestoreEntitiesUsingThisModel()
+{
+    CModelInfo* pModelInfo = g_pGame->GetModelInfo(m_iModelID, true);
```

Comment:
Don't you also need an `if (!m_bAllocatedByUs)` here?

Reply:
I think we can remove m_bAllocatedByUs from this class completely

Reply:
Here can be check. But this is unnecessary for now. This code used only for deallocation.